### PR TITLE
First pass at better URL encoding

### PR DESCRIPTION
**Pull request by samgreen**
Remove nonsensical URL encoding from MPNotification.

Remove MPNetworkTests.h and move to MPNetworkPrivate.h, fix tests since there was a small refactor, add test that checks full URL is built correctly for decide changes.

### CODE REVIEW

File: Mixpanel/MPNetwork.h [hunk 1]
```diff
@@ -20,4 +20,12 @@
 
 - (void)updateNetworkActivityIndicator:(BOOL)enabled;
 
+- (NSURLRequest *)requestForEndpoint:(NSString *)endpoint
```
Comment (patedit) [inline Mixpanel/MPNetwork.h]: If we keep `queryItems` and `andBody` we don't need to pass the http method. Alternatively, we could just keep `byHTTPMethod` and pass an `id` (not ideal, I prefer the first solution).

File: Mixpanel/MPNetwork.m [hunk 2]
```diff
@@ -115,16 +116,62 @@ - (BOOL)handleNetworkResponse:(NSHTTPURLResponse *)response withError:(NSError *
 }
 
 #pragma mark - Helpers
-- (NSURLRequest *)requestForEndpoint:(NSString *)endpoint withBody:(NSString *)body {
-    NSURL *URL = [self.serverURL URLByAppendingPathComponent:endpoint];
-    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
++ (NSArray<NSURLQueryItem *> *)buildDecideQueryForProperties:(NSDictionary *)properties
+                                              withDistinctID:(NSString *)distinctID
+                                                    andToken:(NSString *)token {
+    NSURLQueryItem *itemVersion = [NSURLQueryItem queryItemWithName:@"version" value:@"1"];
```
Comment (patedit) [inline Mixpanel/MPNetwork.m]: Is this a new param? Why do we need it?

File: Mixpanel/MPNetwork.m [hunk 3]
```diff
@@ -115,16 +116,62 @@ - (BOOL)handleNetworkResponse:(NSHTTPURLResponse *)response withError:(NSError *
...
+    NSURLQueryItem *itemLib = [NSURLQueryItem queryItemWithName:@"lib" value:@"iphone"];
```
Comment (patedit) [inline Mixpanel/MPNetwork.m]: `lib` or `mp_lib`?

File: Mixpanel/Mixpanel.m [hunk 4]
```diff
@@ -982,7 +975,9 @@ - (void)applicationDidEnterBackground:(NSNotification *)notification
         dispatch_group_enter(bgGroup);
         NSString *requestData = [MPNetwork encodeArrayForAPI:@[@{@"event": @"Integration", @"properties": @{@"token": @"85053bf24bba75239b16a601d9387e17", @"mp_lib": @"iphone", @"distinct_id": self.apiToken}}]];
         NSString *postBody = [NSString stringWithFormat:@"ip=%d&data=%@", self.useIPAddressForGeoLocation, requestData];
-        NSURLRequest *request = [self.network requestForEndpoint:@"/track/" withBody:postBody];
+        NSURLRequest *request = [self.network requestForEndpoint:@"/track/"
```
Comment (patedit) [inline Mixpanel/Mixpanel.m]: Could we use constants for this and `decide`?

File: Mixpanel/MPNetwork.h (hunk 1 above)
Comment (samgreen) [inline Mixpanel/MPNetwork.h]: Agree, will change.

File: Mixpanel/MPNetwork.m (hunk 2 above)
Comment (samgreen) [inline Mixpanel/MPNetwork.m]: This is an old param. Remember this is for *decide*

File: Mixpanel/MPNetwork.m (hunk 3 above)
Comment (samgreen) [inline Mixpanel/MPNetwork.m]: `lib` we're building for *decide* and not *track*

File: Mixpanel/Mixpanel.m (hunk 4 above)
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: Yes, will do.

Comment (patedit) [inline Mixpanel/Mixpanel.m]: And engage, sorry

**Comment by patedit** [comment]
LGTM

**Pull request merged**